clio-server/src/test/scala/org/broadinstitute/clio/server/dataaccess/HttpElasticsearchDAOSpec.scala

```diff
@@ -105,7 +110,48 @@ class HttpElasticsearchDAOSpec
     } yield succeed
   }
 
-  // TODO: The following tests are demo code, and should be replaced with actual DAO methods/specs.
+  it should "return the most recent document" in {
+    import HttpElasticsearchDAOSpec._
+    import com.sksamuel.elastic4s.circe._
+    import org.broadinstitute.clio.server.dataaccess.elasticsearch.Elastic4sAutoDerivation._
+    import org.broadinstitute.clio.server.dataaccess.elasticsearch._
+
+    val index =
+      new AutoElasticsearchIndex[Document]("docs-" + UUID.randomUUID())
+
+    val documents =
+      (1 to 4)
+        .map("document-" + _)
+        .map(Document(ClioUUIDGenerator.getUUID(), _))
+
+    for {
+      _ <- httpElasticsearchDAO.createOrUpdateIndex(index)
+      _ <- Future.sequence(
+        documents.map(httpElasticsearchDAO.updateMetadata(_, index))
+      )
+      document <- httpElasticsearchDAO.getMostRecentDocument(index)
+      _ = {
+        document should be(documents.last)
+      }
+    } yield succeed
```

Comment:
This can be cleaner by moving the assertion into the `yield` block:
```scala
for {
  ...
  document <- ...
} yield {
  document should be(documents.last)
}
```